src/peer/socket.js

```diff
@@ -169,34 +169,49 @@ class Socket extends EventEmitter {
 
       http.timeout = config.DISPATCHER_TIMEOUT;
       http.open('GET', this._dispatcherUrl, true);
-
       /* istanbul ignore next */
       http.onerror = () => {
-        reject(new Error('There was a problem with the dispatcher.'));
+        reject(
+          new Error(
+            'There was a problem with the request for the dispatcher. Check your request and network connections.'
+          )
+        );
       };
 
       http.onabort = () => {
-        reject(new Error('The request was aborted.'));
+        reject(new Error('The request for the dispatcher was aborted.'));
       };
 
       http.ontimeout = () => {
-        reject(new Error('The request for the dispather timed out.'));
+        reject(
+          new Error(
+            'The request for the dispatcher timed out. Check your firewall, network speed, SkyWay failure information'
+          )
+        );
       };
-
-      http.onreadystatechange = () => {
-        if (http.readyState !== 4) {
-          return;
-        }
-
-        // maybe aborted or something
-        if (http.status === 0) {
-          reject(new Error('There was a problem with the dispatcher.'));
+      http.onload = () => {
+        let res = null;
```

Comment:
This variable should be `const` now and move to inner scope.

Reply:
Thank you, I fixed it